app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,49 @@
+class TripsController < ApplicationController
+  before_action :find_trip_by_id, only: %i[edit update destroy]
+
+  def index
+    @trips = Trip.all.order('id')
+  end
+
+  def new
+    @trip = Trip.new
+  end
+
+  def edit; end
+
+  def create
+    @trip = Trip.new(trip_params)
+    if @trip.save
+      redirect_to action: 'index', status: :created
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @trip.update(trip_params)
+      redirect_to action: 'index', status: :see_other
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @trip.destroy
+    redirect_to action: 'index', status: :see_other
+  end
+
+  private
+
+  def trip_params
+    params.require(:trip).permit(:ticket_price, :total_booked, :date, :time, :bus_id, :route_id)
+  end
+
+  def find_trip_by_id
+    @trip = Trip.find_by(id: params[:id])
+    return if @trip
+
+    flash[:alert] = "Trip #{t(:not_found)}"
```

Comment:
I would recommend to use i18n translation for the error and success messages. This will be a new topic. Feel free to reach out for any help regarding this. Thank you!

Reply:
Thanks for the recommendation apu. Implemented i18n for all the error and success messages. Used two different sets of messages in `en.yml` and `es.yml`. By using `set_locale` method in `Application Controller` , we can changes the language of those messages.